visitations/src/main.rs

```
@@ -0,0 +1,333 @@
+        .await
+        .map_err(HandleError::TypeDB)?
+    {
+        println!("\nSchema Already Defined\n");
+    } else {
+        databases
+            .create(DATABASE)
+            .await
+            .map_err(HandleError::TypeDB)?;
+        let session = Session::new(
+            databases
+                .get(DATABASE)
+                .await
+                .map_err(HandleError::TypeDB)?,
+            Schema,
+        )
+        .await
+        .map_err(HandleError::TypeDB)?;
+        let transaction = session
+            .transaction(Write)
+            .await
+            .map_err(HandleError::TypeDB)?;
+        transaction
+            .query()
+            .define(schema.as_str())
+            .await
+            .map_err(HandleError::TypeDB)?;
+        transaction.commit().await.map_err(HandleError::TypeDB)?;
+        drop(session);
+        println!("\nSchema Defined Successfully\n");
+    }
+
+    Ok(())
+}
+
+fn create_visited_relationship(
+    start: i32,
+    end: i32,
+    first_name: &str,
+    last_name: &str,
+    city: &str,
+    month_id: i32,
+) -> String {
+    format!(
+        r#"match $person isa person, has firstname "{first_name}", has lastname "{last_name}";
+    $city1 isa city, has name "{city}";
+    insert (person: $person, destination: $city1) isa visited, has start_date 2023-0{month_id}-{start}, has end_date 2023-0{month_id}-{end};"#
+    )
+}
+
+async fn create_data(connection: Connection) ->  Result<(), HandleError> {
+    let databases = DatabaseManager::new(connection.clone());
+    // insert data
+    let session = Session::new(databases.get(DATABASE).await.unwrap(), Data)
+        .await
+        .unwrap();
+    let transaction = session.transaction(Write).await.unwrap();
+    let cities = ["Tbilisi", "London", "Cambridge", "Zugdidi"];
+    cities
+        .iter()
+        .map(|city| {
+            let _ = transaction.query().insert(&create_city_entity(city));
+        })
+        .count();
```

Looks nice without `for` loop! But it is still possible to use `.ok()` instead of `let _ =` if we'd like to ignore errors. It is a more conventional way.
But isn't it better not to ignore errors? Here are some ways to iterate over `Result`s: [https://doc.rust-lang.org/rust-by-example/error/iter_result.html](https://doc.rust-lang.org/rust-by-example/error/iter_result.html)